File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <math.h>

bool is_armstrong_number(int number) {
    int sum = 0;
    int temp = number;
    int digits = 0;
    while (temp > 0) { // removes the last digit of temp (the number)
        digits++;
        temp /= 10;
    }
    temp = number;
    while (temp > 0) {
        sum += pow(temp % 10, digits); // temp % 10 gets the last digit of temp
        temp /= 10; // removes the last digit of temp
    }
    return number == sum;
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        int length = (int) (Math.log10(numberToCheck) + 1);
        int sum = 0;
        int temp = numberToCheck;

        while (temp != 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, length);
            temp /= 10;
        }

        return sum == numberToCheck;
    }

}
